packages/fxa-auth-server/test/local/payments/stripe.js

```diff
@@ -145,6 +144,71 @@ describe('StripeHelper', () => {
     sandbox.restore();
   });
 
+  describe('cachedResult', () => {
+    const expectedResult = { status: 'success' };
+
+    const cacheKey = 'testKey';
+    let refreshFunction;
+
+    beforeEach(() => {
+      refreshFunction = sandbox.stub().returns(expectedResult);
+    });
+
+    it('follows read-through caching logic', async () => {
+      // On first call: cache miss, call refresh function, cache the result
+      let actualResult = await stripeHelper.cachedResult(
+        cacheKey,
+        refreshFunction
+      );
+      assert(mockRedis.get.calledOnce);
+      assert(mockRedis.set.calledOnce);
+      assert(refreshFunction.calledOnce);
+      assert.deepEqual(actualResult, expectedResult);
+
+      refreshFunction.resetHistory();
```

Comment:
A little confusing here that `resetHistory` is called on `refreshFunction` but not `mockRedis.get` or `mockRedis.set`.  `assert(refreshFunction.calledOnce)` would tell us that it wasn't called again without the reset.